llvmlite/ir/_utils.py

```diff
@@ -53,3 +53,20 @@ def get_reference(self):
         except AttributeError:
             s = self.__cached_refstr = self._get_reference()
             return s
+
+
+class _HasMetadata(object):
```

Comment:
What's the point of a docstring on a private class? It's not visible unless you're looking at the code and there isn't anything to add to the code anyway.